src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -131,14 +187,7 @@ def _write_binding_constraint_ini(
             # Persist the updated INI content
             self.ini_file.write_ini_file()
         else:
-            terms_dict = {term.id: term for term in terms} if terms else {}
-
-            full_properties = BindingConstraintPropertiesLocal(
-                constraint_name=constraint_name,
-                constraint_id=constraint_id,
-                terms=terms_dict,
-                **properties.model_dump(),
-            )
+            full_properties = BindingConstraintPropertiesLocal(**local_properties.model_dump())
```

Comment:
Perhaps I'm missing something but it looks like you don't have to introduce this object and you can just use local_properties as it is